trunk/source/Library-PEA/src/de/uni_freiburg/informatik/ultimate/lib/pea/CDD.java

```diff
@@ -854,4 +855,77 @@ private String toStringInfix(final Function<CDD, String> funCdd2Str,
 		return sb.toString();
 	}
 
+	/**
+	 * Returns an ArrayList of pairs containing all the Decisions in the CDD. The first element of each pair is a
+	 * Decision<?>, the second element is an int signifying the true child of the decision. If called on CDD.TRUE or

```

Comment:
- What is an int array? 
- This comment should not expain what a true child ist, as this is cdd jargon
- A CDD may not have no childen except it is TRUE or FALSE - do you think there should be handling for invalid CDDs?

Reply:
The function returns a `Pair<Decision,int[]>` , not `Pair<Decision,int>`, so it is unclear to me how that array signifies _the_ true child. I was somewhat confused if I did think of the right concept (i.e., perhaps you dont talk about the CDD children in this comment). 
I guess the function transforms a CDD in a custom data structure (in this case, a list of pairs), that can be used in other places because there, it is more convenient. It is a somewhat (to me) weird structure, in particular the `[0,2]` case. I wonder why it isnt a simple int (0=left,1=right,2="not middle").

The "what happens if there are no children" was phrased weirdly. I agree, it should be clear from the `CDD.TRUE/FALSE` case. 

I would drop the part about the assertion and remove the commented-out assertion.

I am now a little bit confused about the name. How does that List map to a conjunction of decisions?

Reply:
I see, the function maps the Decision to _its true children_ usually one, but in  case of e.g. a range decision there may also be two. But I agree that this part is confusing as it translates the CDD into a strangte intermediate representation and re-assembles the CDD later with some transformations applied (the actual reason of this part). 
Again, as this is a more complcated thing this will not happen in this week so I would prefer to merge and later fix this.

Reply:
Yes, but the comment could be fixed, and perhaps a TODO added.